ecosystem/sep-0012.md

```diff
@@ -16,36 +16,39 @@ This SEP defines a standard way for stellar clients to upload KYC (or other) inf
 
 This SEP was made with these goals in mind:
 
-* interoperability
-* Allow a customer to enter their KYC information to their wallet once and use it across many services without re-entering information manually
-* handle the most common 80% of use cases
-* handle image and binary data
-* support the set of fields defined in [SEP-9](sep-0009.md)
-* support authentication via [SEP-10](sep-0010.md)
-* support the provision of data for [SEP-6](sep-0006.md), [SEP-24](sep-0024.md), [SEP-31](sep-0031.md), and others
-* give customers control over their data by supporting complete data erasure
+- interoperability
+- Allow a customer to enter their KYC information to their wallet once and use it across many services without re-entering information manually
+- handle the most common 80% of use cases
+- handle image and binary data
+- support the set of fields defined in [SEP-9](sep-0009.md)
+- support authentication via [SEP-10](sep-0010.md)
+- support the provision of data for [SEP-6](sep-0006.md), [SEP-24](sep-0024.md), [SEP-31](sep-0031.md), and others
+- give customers control over their data by supporting complete data erasure
 
 To support this protocol an anchor acts as a server and implements the specified REST API endpoints, while a wallet implements a client that consumes the API. The goal is interoperability, so a wallet implements a single client according to the protocol, and will be able to interact with any compliant anchor. Similarly, an anchor that implements the API endpoints according to the protocol will work with any compliant wallet.
 
 ## Prerequisites
 
-* An anchor must define the location of their `KYC_SERVER` or `TRANSFER_SERVER` in their [`stellar.toml`](sep-0001.md). This is how a client app knows where to find the anchor's server. A client app will send KYC requests to the `KYC_SERVER` if it is specified, otherwise to the `TRANSFER_SERVER`.
-* Anchors and clients must support [SEP-10](sep-0010.md) web authentication and use it for all SEP-12 endpoints.
+- An anchor must define the location of their `KYC_SERVER` or `TRANSFER_SERVER` in their [`stellar.toml`](sep-0001.md). This is how a client app knows where to find the anchor's server. A client app will send KYC requests to the `KYC_SERVER` if it is specified, otherwise to the `TRANSFER_SERVER`.
+- Anchors and clients must support [SEP-10](sep-0010.md) web authentication and use it for all SEP-12 endpoints.
 
 ## API Endpoints
 
-* [`GET /customer`](#customer-get): Check the status of a customers info
-* [`PUT /customer`](#customer-put): Idempotent upload of customer info
-* [`DELETE /customer`](#customer-delete): Deletion of customer info
+- [`GET /customer`](#customer-get): Check the status of a customers info
+- [`PUT /customer`](#customer-put): Idempotent upload of customer info
+- [`DELETE /customer`](#customer-delete): Deletion of customer info
+- [`POST /customer`](#customer-post): Set callbacks for a wallet to receive status updates from the anchor
```

Comment:
❗I think this breaks the existing REST  pattern this SEP exposes. The description indicates that this sets a value which implies to me it can be called with an updated value. For consistency with the rest of this SEP it should use the `PUT`.

❓Is there a need to set the callback in a separate call? I think the new `on_change_callback` field should be added added to the existing `PUT /customer`. If there's a need to set the callback separately then we should make it a sub-path, e.g. `PUT /customer/callbacks`.

Reply:
I think the POST verb applies better than the PUT verb. PUT implies idempotency, but a wallet may want to set more than one callback on a given KYC user.

I don't feel strongly about `/customer` vs. `/customer/callbacks`, and I think the distinction is a semantic / aesthetic one, so I'll just take your suggestion and change the URL.

Reply:
Is the reason there would be multiple because an account holder might be using several different wallet applications, and each might register for callbacks?

I'd expect if it's just one application it'd be one callback url and if it a new url is generated updating would be sufficient, and so PUT would be more appropriate.

Could you elaborate on the multiple use case?

If we have multiple we need to think about deleting or unregistering callbacks. That probably means returning a callback id. Or define when they expire naturally. Exposing might be better so that old wallets don't get callbacks after a user switches wallets.

If we require the callback to be included in the main PUT call we could tie cannabis to that specific kyc run, as opposed to being registered with the account forever. Once that kyc run completes it could be considered expired.

Reply:
Is the reason there would be multiple because an account holder might be using several different wallet applications, and each might register for callbacks?

I'd expect if it's just one application it'd be one callback url and if it a new url is generated updating would be sufficient, and so PUT would be more appropriate.

Could you elaborate on the multiple use case?

If we have multiple we need to think about deleting or unregistering callbacks. That probably means returning a callback id. Or define when they expire naturally. Exposing might be better so that old wallets don't get callbacks after a user switches wallets.

If we require the callback to be included in the main PUT call we could tie cannabis to that specific kyc run, as opposed to being registered with the account forever. Once that kyc run completes it could be considered expired.

Reply:
Returning to a question I asked that wasn't answered:
>Is the reason there would be multiple because an account holder might be using several different wallet applications, and each might register for callbacks?

I think this is a valid use case, and so it would have also been fine to support multiple callbacks but clean up the REST pattern in use to support registering a callback and deleting it. Example:
- `POST /customer/callbacks` returns a callback ID
- `DELETE /customer/callbacks/<id>` deletes the callback (optional given that callbacks stop once the account arrives at a terminal state).

If a single callback is all we need in the foreseeable future then we don't need to add the burden of multiple callbacks to SEP-12 implementations, and we can always add other endpoints when they're needed at a later date.